geps/gep-1897.md

```diff
@@ -125,7 +125,231 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+be validated by the provided trusted CA certificates before the connection is made.  On the question of how to signal
+that there was a failure in the certificate validation, this is left up to the implementation to return a response error
+that is appropriate, such as one of the HTTP error codes: 400 (Bad Request), 401 (Unauthorized), 403 (Forbidden), or
+other signal that makes the failure sufficiently clear to the requester without revealing too much about the transaction,
+based on established security requirements.
+
+Not covered here, but possible to add would be additional configuration options mentioned in [SIG-NET Gateway API: TLS
+to the K8s.Service/Backend](https://docs.google.com/document/d/1RTYh2brg_vLX9o3pTcrWxtZSsf8Y5NQvIG52lpFcZlo).  All of
+these are currently implementation-dependent, with the following recommended defaults:
+- Server Name Indication: enables passing of the server name through server name indication, in the TLS transaction, to
+assist with selection of certificates when several hosts share the same IP address. (default to enabled)
+- Subject Alternative Name certificates: enable the use of a single certificate that can serve multiple domains.
+(default to enabled)
+- Version: specifies the minimum TLS version that the connection may use (default TLSv1.2)
+- Ciphers: specifies enabled ciphers to be used in TLS exchanges. (default to align with TLS versions 1.1, 1.2, and/or
+1.3 as described in [RFC4346 TLS 1.1](https://datatracker.ietf.org/doc/html/rfc4346#appendix-A.5),
+[RFC 5246 TLS 1.2](https://datatracker.ietf.org/doc/html/rfc5246#appendix-A.5), and
+[RFC 8446 TLS 1.3](https://datatracker.ietf.org/doc/html/rfc8446#appendix-B.4).
+- Version: specifies the minimum TLS version that the connection may use (default TLSv1.2)
+- Visibility: the visibility level could be all or none, indicating that if the connection failed due to validation
+failures, it would drop the connection silently if the visibility level were none, and report an error if the visibility
+level were all.  (default to all)
+
+```go
+// TLSConnectionPolicy provides a way to publish TLS configuration
+// that enables a gateway client to connect to a backend pod.
+type TLSConnectionPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of TLSConnectionPolicy.
+    Spec TLSConnectionPolicySpec `json:"spec"`
+
+    // Status defines the current state of TLSConnectionPolicy.
+    Status TLSConnectionPolicyStatus `json:"status,omitempty"`
+}
+
+// TLSConnectionPolicySpec defines the desired state of
+// TLSConnectionPolicy.
+// Note: there is no Override or Default policy configuration.
+type TLSConnectionPolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
```

Comment:
I'm really interested in if we can/should extend the TargetRef in the policy attachment GEP to optionally include port. It feels really strange to have port outside of the TargetRef because it's really just targeting a subset of that resource.

Reply:
I like the term `section` that can be used to refer to a `listener` name within Gateway, `rule` name within xRoute, `port` name within Service

Reply:
I'd back extending TargetRef in the policy attachment GEP to optionally include port.

Reply:
Maybe what we need to say is:

1. All policy resources must include TargetRef with the fields specified here: https://github.com/kubernetes-sigs/gateway-api/blob/a33a934af9ec6997b34fd9b00d2ecd13d143e48b/apis/v1alpha2/policy_types.go#L24-L41
2. Policy resources _may_ also choose to include `SectionName` and/or `Port` in TargetRef following the same mechanics of `ParentRef`

Then we can provide additional `TargetRef` types that extend the base `TargetRef` with the different possible combinations of these fields. That feels like it could be handled in a separate smaller update to the policy GEP.

Reply:
I added an issue for this: https://github.com/kubernetes-sigs/gateway-api/issues/2147

Reply:
I will add a small mention in the GEP, as well.

Reply:
Thanks @candita! I think this can mostly be resolved since Port has been removed from the GEP and we have a follow up issue for this. Do you mind adding some kind of AI/follow up in this GEP that links to this issue and provides a bit of context about this policy targeting the entirety of a Service?

Reply:
Added to Open Questions section.